corporate/lib/stripe.py

```diff
@@ -1276,49 +1278,65 @@ def create_card_update_session(self) -> Dict[str, Any]:
     def apply_discount_to_plan(
         self,
         plan: CustomerPlan,
-        discount: Decimal,
+        customer: Customer,
     ) -> None:
-        plan.discount = discount
-        plan.price_per_license = get_price_per_license(plan.tier, plan.billing_schedule, discount)
-        plan.save(update_fields=["discount", "price_per_license"])
+        original_plan_price = get_price_per_license(plan.tier, plan.billing_schedule)
+        plan.price_per_license = get_price_per_license(plan.tier, plan.billing_schedule, customer)
+
+        # For display purposes only.
+        plan.discount = format_discount_percentage(
+            Decimal((original_plan_price - plan.price_per_license) / original_plan_price * 100)
+        )
+        plan.save(update_fields=["price_per_license", "discount"])
```

Comment:
Would it make sense to rework `plan.discount` to be a property, rather than a database field? Just thinking about how if we change the base pricing, we'll need to run a migration to update these cached values.

Reply:
Yeah, especially since annual and monthy discounted prices could be different and hence the plan.discount.